src/common/exif.cc

```diff
@@ -1181,13 +1181,33 @@ static bool _exif_decode_exif_data(dt_image_t *img, Exiv2::ExifData &exifData)
     /*
      * Get the focus distance in meters.
      */
-    if(FIND_EXIF_TAG("Exif.NikonLd2.FocusDistance")
-       || FIND_EXIF_TAG("Exif.NikonLd3.FocusDistance")
-       || (Exiv2::testVersion(0, 27, 4)
-           && FIND_EXIF_TAG("Exif.NikonLd4.FocusDistance")))
+    if(Exiv2::testVersion(0, 27, 4) && FIND_EXIF_TAG("Exif.NikonLd4.LensID") && pos->toLong() != 0)
+    {
+      // Z lens, need to specifically look for the second instance of Exif.NikonLd4.FocusDistance
+      // unless using Exiv2 0.28.x and later (also expanded to 2 bytes of precision since 0.28.1).
+#if EXIV2_TEST_VERSION(0, 28, 0)
```

Comment:
Does this test - at compile time - work appropriately, if libexiv2 is linked dynamically as a shared object? What will happen if the user has a version of libexiv2.so on his PC that differs from the version at you build machine?

Reply:
> What will happen if the user has a version of libexiv2.so on his PC that differs from the version at your build machine?

Something like this: https://github.com/darktable-org/darktable/issues/15027

Reply:
Ok. Thanks for the clarification. I don't have any further comments.
Unfortunately, I was not able to test it on my Ubuntu system because I have mixed up my exiv2 installation. So we have to rely on other test reports...
Thank you again for fixing this so fast.